py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -441,3 +254,229 @@ def _request(self, method, url, body=None):
         finally:
             LOGGER.debug("Finished Request")
             resp.close()
+            # clear connections - we lose the connections pools,
+            # but we keep the main object around avoiding recreating it.
+            # and we are avoiding gc cleaning up our sockets
+            if not self.keep_alive:
+                self._conn.clear()
+
+    # now can use with closing!
+    # don't rely on gc to clean things up for you
+    def close(self):
+        self._conn.clear()
+
+    _commands = {
```

Comment:
This seems like a good move, moving the definition to the class body so it's define once and not on every function invocation.